src/main/java/com/northconcepts/datapipeline/examples/cookbook/ReadACsvFileUsingMultithreading.java

```diff
@@ -0,0 +1,36 @@
+package com.northconcepts.datapipeline.examples.cookbook;
+
+import com.northconcepts.datapipeline.core.AsyncReader;
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.DataWriter;
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.examples.database.DB;
+import com.northconcepts.datapipeline.jdbc.JdbcWriter;
+import com.northconcepts.datapipeline.job.Job;
+
+import java.io.File;
+import java.sql.Connection;
+
+public class ReadACsvFileUsingMultithreading {
+    public static void main(String[] args) throws Throwable {
+        DataReader reader = new CSVReader(new File("example/data/input/credit-balance-01.csv"))
+                .setFieldNamesInFirstRow(true);
+        reader = new AsyncReader(reader).setMaxBufferSizeInBytes(1024 * 1024 * 10);
+
+        DB db = new DB();
+        Connection connection = db.getConnection();
```

Comment:
It looks like we get the connection before we are ready to use it.  Let's move this call down to where it's really needed.

Reply:
done